hack/central-repo/upload-plugins.sh

```diff
@@ -0,0 +1,140 @@
+		"Version"            TEXT NOT NULL,
+		"Hidden"             INTEGER NOT NULL,
+		"Description"        TEXT NOT NULL,
+		"Publisher"          TEXT NOT NULL,
+		"Vendor"             TEXT NOT NULL,
+		"OS"                 TEXT NOT NULL,
+		"Architecture"       TEXT NOT NULL,
+		"Digest"             TEXT NOT NULL,
+		"URI"                TEXT NOT NULL,
+		PRIMARY KEY("PluginName", "Target", "Version", "OS", "Architecture")
+	);
+EOF
+
+addPlugin() {
+    name=$1
+    target=$2
+
+    tmpPluginPhase1="/tmp/fakeplugin1.sh"
+    tmpPluginPhase2="/tmp/fakeplugin2.sh"
+
+    # Start preparing the plugin file with the name and target.
+    # Start here to avoir repeating in the loop.
+    sed -e "s/__NAME__/$name/" -e "s/__TARGET__/$target/" $ROOT_DIR/fakeplugin.sh > $tmpPluginPhase1
+
+    # Define 10 versions for the plugin
+    for v in {0..9}; do
+        version="v$v.$v.$v"
+
+        # Put printout to show progress
+        echo "Inserting $name version $version for target $target"
+
+        # Create the plugin file with the correct version
+        sed -e "s/__VERSION__/$version/" $tmpPluginPhase1 > $tmpPluginPhase2
+        digest=$(sha256sum $tmpPluginPhase2 | cut -f1 -d' ')
+
+        for os in darwin linux windows; do
+            for arch in amd64 arm64; do
+                if [ $arch = arm64 ] && [ $os != darwin ]; then
+                    # Only support darwin with arm64 for now
+                    continue
+                fi
+
+                image_path=$publisher/$os/$arch/$target/$name
+
+                sql_cmd="INSERT INTO PluginBinaries VALUES('$name','$target','v9.9.9','$version','FALSE','Desc for $name','TKG','VMware','$os','$arch','$digest','$image_path:$version');"
+                if [ "$dry_run" = "echo" ]; then
+                    echo $sql_cmd 
+                else 
+                    echo $sql_cmd | sqlite3 -batch $database
+                fi
+
+                # For efficiency, only push the plugin binaries of the two latest versions for 4 plugins
+                # The plugins are: twotargets (for both targets), kubeplugin, tmcplugin 
+                if [ $name = "twotargets" ] || [ $name = "kubeplugin" ] || [ $name = "tmcplugin" ]; then
+                    if [ $version = "v8.8.8" ] || [ $version = "v9.9.9" ]; then
+                        ${dry_run} imgpkg push -i $repoBasePath/$image_path:$version -f $tmpPluginPhase2 --registry-insecure
+                    fi
+                fi
+            done
+        done
+    done
+}
+
+# Push the plugins that have two targets
+addPlugin twotargets tmc
+addPlugin twotargets k8s
+
+# Push another two plugins for the small repo
+addPlugin kubeplugin k8s
+addPlugin tmcplugin tmc
+
+# Push 95 more plugins, for a total of 99
+if [ $fast = "off" ]; then
+    for idx in {0..94}; do
+        target_rand=$(($RANDOM % 3))
+        case $target_rand in
+        0) target=global
+           ;;
+        1) target=k8s
+           ;;
+        2) target=tmc
+           ;;
```

Comment:
Just to clarify, we are storing all non-targeted plugins with Target value as `global` in the database. Correct? 
Basically `TARGET` field within the database will have either a `global`, `k8s`, or `tmc` value and it should not be empty.

Reply:
I'm glad you caught that 😄 .  This is something I needed to discuss with you to align with the publishing tooling.
I used the "global" keyword to make it easier to read the database manually, but in the code I convert it back to the empty string to fit with the code base.  I am fine with leaving the field empty here.  What do you prefer?

I wanted to confirm with you that we don't support a plugin in the "global" target AND the same plugin name in the "tmc" target?  The only possible double target plugin name is "k8s" and "tmc", right?

Reply:
> I wanted to confirm with you that we don't support a plugin in the "global" target AND the same plugin name in the "tmc" target? 

We don't support a plugin in the "global" target AND the same plugin name in the **"k8s"** target.

The below combinations are supported:

```
name: plugin0 , target: global
name: plugin0 , target: tmc
```
```
name: plugin1 , target: k8s
name: plugin1 , target: tmc
```

The below combination is NOT supported:
```
name: plugin2 , target: k8s
name: plugin2 , target: global
```

Reply:
> I used the "global" keyword to make it easier to read the database manually, but in the code I convert it back to the empty string to fit with the code base. I am fine with leaving the field empty here. What do you prefer?

I am fine with using "global" keyword in the database.

Reply:
> The below combinations are supported:
>
> name: plugin0 , target: global
> name: plugin0 , target: tmc

@anujc25 That's the one I've been thinking about.  How would a user install plugin0 for the global target?
`tanzu plugin install plugin0` will find two plugins but we don't have a target name for the global target to use with the `--target` flag